astropy/nddata/utils.py

```diff
@@ -456,3 +469,270 @@ def block_replicate(data, block_size, conserve_sum=True):
+            cutout array that do not overlap with the input ``data``.
+            ``fill_value`` must have the same ``dtype`` as the input
+            ``data`` array.
+
+        Returns
+        -------
+        result : `~astropy.nddata.utils.Cutout2D`
+            A cutout object containing the 2D cutout data array and the
+            updated WCS, if ``wcs`` is input.
+
+        Examples
+        --------
+        >>> import numpy as np
+        >>> from astropy.nddata.utils import Cutout2D
+        >>> data = np.arange(20.).reshape(5, 4)
+        >>> c1 = Cutout2D(data, (2, 2), (3, 3))
+        >>> print(c1.data)
+        [[  5.   6.   7.]
+         [  9.  10.  11.]
+         [ 13.  14.  15.]]
+
+        >>> print(c1.center_large)
+        (2.0, 2.0)
+        >>> print(c1.center_small)
+        (1.0, 1.0)
+        >>> print(c1.origin_large)
+        (1, 1)
+
+        >>> c2 = Cutout2D(data, (0, 0), (3, 3))
+        >>> print(c2.data)
+        [[ 0.  1.]
+         [ 4.  5.]]
+
+        >>> c3 = Cutout2D(data, (0, 0), (3, 3), mode='partial')
+        >>> print(c3.data)
+        [[ nan  nan  nan]
+         [ nan   0.   1.]
+         [ nan   4.   5.]]
+        """
+
+        if isinstance(position, SkyCoord):
+            if wcs is None:
+                raise ValueError('wcs must be input if position is a '
+                                 'SkyCoord')
+            position = skycoord_to_pixel(position, wcs, mode='all')
+
+        if mode not in ['partial', 'trim', 'strict']:
+            raise ValueError("Valid modes are 'partial', 'trim', and "
+                             "'strict'.")
+
+        data = np.asanyarray(data)
+        self.data, self.input_position_small  = extract_array(data, shape,
+            position, mode=mode, fill_value=fill_value, return_position=True)
+
+        slices_large, slices_small = overlap_slices(data.shape, shape,
+                                                    position, mode=mode)
+        self.slices_large = slices_large
+        self.slices_small = slices_small
+
+        self.shape = self.data.shape
+        self.input_position_large = position
+        self.shape_input = shape
+
+        (self.ymin_large, self.xmin_large,
+         self.ymax_large, self.xmax_large) = self.bbox_large
+
+        (self.ymin_small, self.xmin_small,
+         self.ymax_small, self.xmax_small) = self.bbox_small
+
+        # the true origin pixel of the cutout array, including any
+        # filled cutout values
+        self._origin_large_true = (self.origin_large[1] -
+                                   self.slices_small[1].start,
+                                   self.origin_large[0] -
+                                   self.slices_small[0].start)
+
+        if wcs is not None:
+            self.wcs = copy.deepcopy(wcs)
+            self.wcs.wcs.crpix -= self._origin_large_true
+        else:
+            self.wcs = None
+
+    def to_large(self, position):
```

Comment:
Thanks, @astrofrog.  The methods are now `to_original_position()` and `to_cutout_position()`.  I also renamed everything from "large" to "original" and "small" to "cutout".